openedx/adg/lms/applications/management/commands/tests/test_update_is_prerequisite_courses_passed.py

```diff
@@ -0,0 +1,177 @@
+"""
+Tests to check if `update_is_prerequisite_courses_passed` command actually updates the `is_prerequisite_courses_passed`
+flag in Application Hub for each eligible user
+"""
+
+from datetime import datetime
+
+import pytest
+from django.core.management import call_command
+
+from openedx.adg.common.course_meta.models import CourseMeta
+from openedx.adg.common.course_meta.tests.factories import CourseMetaFactory
+from openedx.adg.lms.applications.management.commands import update_is_prerequisite_courses_passed as command_module
+from openedx.adg.lms.applications.models import ApplicationHub
+from openedx.core.djangoapps.content.course_overviews.tests.factories import CourseOverviewFactory
+from student.tests.factories import CourseEnrollmentFactory, UserFactory
+
+
+@pytest.fixture
+def user_for_testing(request):
+    return UserFactory.create()
+
+
+@pytest.fixture
+def course_for_testing(request):
+    prerequisite_course = CourseOverviewFactory.create(
+        end_date=datetime(2022, 1, 1),
+        start_date=datetime(2015, 1, 1),
+        display_name='pre requisite course'
+    )
+
+    CourseMetaFactory(course=prerequisite_course, is_prereq=True)
+
+    return prerequisite_course
+
+
+@pytest.fixture
+def enroll_user(request, user_for_testing, course_for_testing):
+    CourseEnrollmentFactory.create(
+        user=user_for_testing,
+        course_id=course_for_testing.id,
+        is_active=request.param['is_active']
+    )
+
+
+@pytest.mark.django_db
+@pytest.mark.parametrize('enroll_user', [dict({'is_active': False})], indirect=True)
+def test_minimal_users_with_inactive_user(enroll_user):
+    """
+    Test to check that an inactive user is not counted while calculating minimal users
+    """
+    pre_req_courses = CourseMeta.open_pre_req_courses.all()
+
+    command = command_module.Command()
+    minimal_users = command.get_minimal_users_to_be_checked_for_update(pre_req_courses)
+
+    assert minimal_users.count() == 0
+
+
+@pytest.mark.django_db
+def test_minimal_users_none_registered_in_prerequisite(course_for_testing):
+    """
+    Test to check that a user not registered in prerequisite is not counted while calculating minimal users
+    """
+    pre_req_courses = CourseMeta.open_pre_req_courses.all()
+
+    command = command_module.Command()
+    minimal_users = command.get_minimal_users_to_be_checked_for_update(pre_req_courses)
+
+    assert minimal_users.count() == 0
+
+
+@pytest.mark.django_db
+@pytest.mark.parametrize('enroll_user', [dict({'is_active': True})], indirect=True)
+def test_minimal_users(enroll_user):
+    """
+    Test to check that an active user registered in prerequisite is counted while calculating minimal users
+    """
+    pre_req_courses = CourseMeta.open_pre_req_courses.all()
+
+    command = command_module.Command()
+    minimal_users = command.get_minimal_users_to_be_checked_for_update(pre_req_courses)
+
+    assert minimal_users.count() > 0
```

Comment:
Please add test cases for the following scenarios:

Create two users, user 1 and user 2, leave user 1 as-is and wherever required, enroll user 2 to course(s).

Create three courses of them marked two pre-req

1. Enroll in two courses, enrollment active in both, and pass both courses as param (list) `pre_req_courses` to `get_minimal_users_to_be_checked_for_update`. Test should user 2 

2. Enroll in three courses, enrollment active in two, and pass both courses as param (list) `pre_req_courses` to `get_minimal_users_to_be_checked_for_update`. Test should user 2 

3. Enroll in two courses, enrollment active in two, and pass two courses (one in which you enrolled and one in which you did not enroll) as param (list) `pre_req_courses` to `get_minimal_users_to_be_checked_for_update`. Test should not return any user

4. Enroll in two courses, enrollment active in two, and pass two courses (one in which you enrolled and one in which you did not enroll) as param (list) `pre_req_courses` to `get_minimal_users_to_be_checked_for_update`. Test should not return any user

5. Enroll in two courses, enrollment active in both, and pass all three courses as param (list) `pre_req_courses` to `get_minimal_users_to_be_checked_for_update`. Test should not return any user

6. In application hub set `is_prerequisite_courses_passed=true`, Enroll in two courses, enrollment active in both, and pass both courses as param (list) `pre_req_courses` to `get_minimal_users_to_be_checked_for_update`. Test should not return any user.